src/client/Microsoft.Identity.Client/ManagedIdentity/ImdsManagedIdentitySource.cs

```diff
@@ -305,8 +305,9 @@ public static async Task<bool> ProbeImdsEndpointAsync(
             }
             catch (Exception ex)
             {
-                requestContext.Logger.Info($"[Managed Identity] {imdsStringHelper} probe endpoint failure. Exception occurred while sending request to probe endpoint: {ex}");
-                return false;
+                throw new MsalServiceException(
```

Comment:
How about this: 

1. Let cancellation exceptions pass through
2. Capture the messages why IMDS1 and IMDS2 probe failed and output them when app calls AcquireToken() . Smth like "Cannot acquire a token for managed identity. IMDSv1 discovery failed with xyz and IMDSv2 discovery failed with bar"

Reply:
If we are able to pass the failure reason back to the caller, it would be better to have it at GetSourceAsync than at GetToken. For DefaultAzureCredential, it will determine if we skip managed identity alltogether.